factory-method/src/main/java/com/iluwatar/factory/method/ElfBlacksmith.java

```diff
@@ -22,16 +22,25 @@
  */
 package com.iluwatar.factory.method;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * 
  * Concrete subclass for creating new objects.
  * 
  */
 public class ElfBlacksmith implements Blacksmith {
 
+  private static Map<WeaponType, ElfWeapon> ELF_ARSENAL = new HashMap<>(WeaponType.values().length);
+  {
+    for (WeaponType type : WeaponType.values()) {
```

Comment:
You can use Java 11 local variable type inference here (var)

Reply:
Nope we can't as var is not applicable for method/constructor/catch formals, method return types, instance variables, or any other variable declaration

reference @ https://dzone.com/articles/java-10-lets-play-with-var